compiler/src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -358,9 +358,13 @@ object Erasure {
     /** This override is only needed to semi-erase type ascriptions */
     override def typedTyped(tree: untpd.Typed, pt: Type)(implicit ctx: Context): Tree = {
       val Typed(expr, tpt) = tree
-      val tpt1 = promote(tpt)
-      val expr1 = typed(expr, tpt1.tpe)
-      assignType(untpd.cpy.Typed(tree)(expr1, tpt1), tpt1)
+      val tpt1 = tpt match {
+        case Block(_, tpt) => tpt // erase type aliases (statements) from type block
+        case tpt => tpt
+      }
```

Comment:
The problematic `Typed.tpt` is inserted in [TypeTestsCasts.scala#L236](https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/transform/TypeTestsCasts.scala#L236), this case could also be handled there.